### ISSUE TITLE
Orphan `sst dev` processes

### ISSUE THREAD

**Issue by jayair** [sst/ion#76]
`sst dev` seems to get fail silently if there's a orphan process running. Can we detect it and throw an error?

https://discord.com/channels/983865673656705025/1177071497974648952/1221876710820151450

<sub>[ION-239](https://linear.app/sst-dev/issue/ION-239/orphan-sst-dev-processes)</sub>

**Comment by AlessandroVol23** [comment]
Not sure if connected but I had a same issue here: https://discord.com/channels/983865673656705025/1177071497974648952/1222858865515692162

An SST V2 app was running and I couldn't start `sst dev` 
The output was the following:

```
time=2024-03-28T14:06:17.782+01:00 level=INFO msg="checking for pulumi" path="/Users/alessandrovolpicella/Library/Application Support/sst/bin/pulumi"
time=2024-03-28T14:06:18.034+01:00 level=INFO msg="checking for bun" path="/Users/alessandrovolpicella/Library/Application Support/sst/bin/bun"
time=2024-03-28T14:06:18.034+01:00 level=INFO msg=args args=[] length=0
time=2024-03-28T14:06:18.035+01:00 level=INFO msg="no existing server found, starting new one"
time=2024-03-28T14:06:18.036+01:00 level=INFO msg="waiting for server to start"
```

**Comment by VirtuaBoza** [comment]
I am currently experiencing `sst dev` exit/fail silently, and I _don't_ have an orphaned sst process running. I just restarted my machine to be sure.

**Comment by VirtuaBoza** [comment]
`sst remove` and then `sst dev` seemed to resolve this, but ovbs i can't do that every time this happens.

**Comment by VirtuaBoza** [comment]
> `sst remove` and then `sst dev` seemed to resolve my issue here, but ovbs i can't do that every time this happens.

Not a reliable remedy it turns out.

**Comment by VirtuaBoza** [comment]
here's a facsimile of my repo: https://github.com/VirtuaBoza/sst-repro-2

**Comment by VirtuaBoza** [comment]
[sst.log](https://github.com/sst/ion/files/14797711/sst.log)
[event.log](https://github.com/sst/ion/files/14797712/event.log)

**Comment by thdxr** [comment]
i think we've fixed this in 0.0.240 but it might still happen in other situations - post a new issue if you see it
